### PR TITLE
Reformulate domains to improve sparsity

### PR DESCRIPTION
**Pull request by hwpang**
See #54.

This PR integrating pressure or volume (when not constant or parametrized) into equations for ConstantVDomain, ParameterizedVDomain, ParameterizedTPDomain, ConstantPDomain and ParameterizedPDomain.

### CODE REVIEW

File: src/TestReactors.jl [hunk 1]
```diff
@@ -10,7 +10,7 @@ spcs = phaseDict["phase"]["Species"]; #mechanism dictionaries index:  phaseDict[
 rxns = phaseDict["phase"]["Reactions"];
 solv = phaseDict["Solvents"][1];
 liq = IdealDiluteSolution(spcs,rxns,solv;name="phase",diffusionlimited=true) #Define the phase (how species thermodynamic and kinetic properties calculated)
-initialconds = Dict(["T"=>450.0,"P"=>1e5,"V"=>1.0e-6*1e6,"octane"=>6.154e-3*1e6,"oxygen"=>4.953e-6*1e6]) #Set simulation Initial Temp and Pressure
```
Comment (mjohnson541) [inline src/TestReactors.jl:13]: Can you check if this needs fixed in any of the ipython notebook examples for liquids (ConstantTVDomain and ParameterizedTConstantVDomain)?

File: src/Domain.jl [hunk 2]
```diff
@@ -1485,6 +1485,7 @@ end
 
 @inline function calcdomainderivatives!(d::ParametrizedVDomain{W,Y},dydt::K,interfaces::Z12;t::Z10,T::Z4,P::Z9,Us::Z,Hs::Z11,V::Z2,C::Z3,ns::Z5,N::Z6,Cvave::Z7) where {Z11,Z10,Z9,W<:IdealGas,Z7,K,Y<:Integer,Z6,Z,Z2,Z3,Z4,Z5,Z12}
     @views @fastmath @inbounds dydt[d.indexes[3]] = (-dot(Us,dydt[d.indexes[1]:d.indexes[2]])-P*Calculus.derivative(d.V,t))/(N*Cvave) #divide by V to cancel ωV to ω
+    @views @fastmath @inbounds dydt[d.indexes[4]] = sum(dydt[d.indexes[1]:d.indexes[2]])*R*T/V + dydt[d.indexes[3]]*P/T - P/V*Calculus.derivative(d.V,t)
```
[the remark this answers is not in the document]
Comment (hwpang) [reply src/Domain.jl]: Thanks for the comment! I changed the code.

File: src/Simulation.jl [hunk 3]
```diff
@@ -61,6 +61,7 @@ getV(bsol::Simulation{Q,W,L,G}, t::K) where {W<:Union{ConstantTPDomain,ConstantP
 getV(bsol::Simulation{Q,W,L,G}, t::K) where {W<:Union{ConstantVDomain,ConstantTVDomain,ParametrizedTConstantVDomain},K<:Real,Q,G,L} = bsol.domain.V
```
[the remark this answers is not in the document]
Comment (hwpang) [reply src/Simulation.jl:61]: I removed the old getV of ConstantPDomain. Thanks!

File: src/TestReactors.jl (hunk 1 above)
Comment (hwpang) [reply src/TestReactors.jl:13]: I removed the initialconds P from ConstantTVDomain in IJulia example. The ParametrizedTConstantVDomain IJulia example doesn't have P in its initialconds.

**Comment by hwpang** [comment]
@mjohnson541 Thanks for your comments! I modified the code as suggested.

File: src/Domain.jl [hunk 4]
```diff
@@ -41,7 +41,7 @@ function ConstantTPDomain(;phase::E2,initialconds::Dict{X,X2},constantspecies::A
     #set conditions and initialconditions
     T = 0.0
     P = 0.0
-    y0 = zeros(length(phase.species)+1)
```
Comment (mjohnson541) [inline src/Domain.jl:44]: I think this commit reverting integrateV got caught unintentionally, it should be deleted.

File: src/Domain.jl [hunk 5]
```diff
@@ -480,7 +480,7 @@ function ConstantTVDomain(;phase::Z,initialconds::Dict{X,E},constantspecies::Arr
         if key == "T"
             T = val
         elseif key == "P"
-            P = val
+            throw(error("ConstantTVDomain cannot specify P"))
```
Comment (mjohnson541) [inline src/Domain.jl:483]: Should we do this in ParameterizedPConstantVDomain as well?

File: src/Domain.jl (hunk 4 above)
Comment (hwpang) [reply src/Domain.jl]: Thanks for the comment. I dropped the reverting commit!

File: src/Domain.jl (hunk 5 above)
Comment (hwpang) [reply src/Domain.jl:484]: This Domain already has an input check for P in the later part. I moved the checking code a bit forward.

**Pull request merged**